lib/SIL/Utils/Dominance.cpp

```diff
@@ -141,3 +141,131 @@ void PostDominanceInfo::verify() const {
     abort();
   }
 }
+
+void swift::partitionByDominance(DominanceInfo &dominance,
```

Comment:
I'm not really happy with this API. It is a complicated algorithm and can be quadratic in worst case.
I think there is a better way to solve this problem (see my other comment) - and other problems which would use this API.